### DISCUSSION TITLE
Como praticar novas tecnologias?

### DISCUSSION THREAD

**Discussion by kvnol** [frontendbr/forum#1305] [Discussão]
Sabemos que só aprendemos, verdadeiramente, com a mão na massa. Em cursos entendemos muito das teorias, mas a organização dos professores de cursos são programadas para dar tudo certo no final.

Nos cursos não passamos as loucuras do dia-a-dia do desenvolvimento de software, por isso a galera que está começando passa muito tempo em agência fazendo e-mail marketing e landing page, pois não tem experiência com desenvolvimento de software e não consegue responder numa entrevista que tem real conhecimento num Angular da vida.

Agora eu te pergunto <del>(leia com a voz do Marcelo Rezende)</del>, como praticar projetos reais, dificuldades reais, sem ter experiência (e não conseguir experiência por não ter experiência também)?

**Comment by Gpaula95** [comment]
Eu particularmente procurei projetos opensource ou até mesmo trabalho voluntario em ongs que tem sistemas mais atualizados. Isso não vai te deixar rico, porem vai te dar uma base muito legal pois você vai encarar projetos reais que outras pessoas deixaram la largados (as vezes pode ser um baita b.o). Outra forma interessante de adquirir experiencia é criando seu proprio produto, com certeza você tem ideias inovadoras guardadas na sua cabeça, Coloque em pratica ! 

Da um fork em um projeto, brinca com ele, resolva bugs e crie novas features, isso vai ajudar as pessoas e vai te dar uma baita base.

PS. PR's não precisam ser aceitas então não tenha medo de errar ;)

**Comment by Gpaula95** [comment]
Caso tenha interesse, tenho contato com algumas ongs que precisam.

**Comment by felipefialho** [comment]
Anos atrás [escrevi um artigo](https://www.felipefialho.com/blog/2015/aproximando-seus-estudos-de-cenarios-reais) falando justamente sobre isso e mantenho a mesmíssima opinião.

Participar de projetos open-source, colocando prazos e "pressão" para si mesmo, é uma ótima maneira de praticar e se testar.

Não existe forma melhor de aprendizado do que mão na massa.

**Comment by simaodeveloper** [comment]
@LFeh Obrigado estava precisando ler isso.

as vezes fico parasitando achando que as coisas vão cair do céu, mas o que precisamos mesmo é colocar a mão na massa e se aproximar mais de um prazo real, alterações de clientes fictícios e coisas assim.

Valeu 😃

**Comment by felipefialho** [comment]
Esse texto tem uns 4 anos, então não reparem nos possíveis erros... haha

**Comment by simaodeveloper** [comment]
@Gpaula95 isso muito me interessa!

**Comment by ericvvf** [comment]
>
> Não existe forma melhor de aprendizado do que mão na massa.

Na linha do que falou o @LFeh:
[The secret to being a top developer is building things! Here’s a list of fun apps to build!](https://medium.freecodecamp.org/the-secret-to-being-a-top-developer-is-building-things-heres-a-list-of-fun-apps-to-build-aac61ac0736c)
[Want to be a top developer? You should build things. Here’s another list to get you started.](https://medium.freecodecamp.org/the-secret-to-being-a-top-developer-is-building-things-d3d058e4e472)

Já pus como meta em 2019: testar novas coisas construindo alguns destes apps!

**Comment by kakastefane** [comment]
Muito boas dicas do @LFeh e a meta do @ericOnda.

Já costumo desenvolver pequenas coisas que preciso no dia-a-dia no jsfiddle, realmente, e ajuda bastante nos estudos mesmo.

**Comment by ninetails** [comment]
> Esse texto tem uns 4 anos, então não reparem nos possíveis erros... haha

ainda está bem atual rs

Bom, não sou daqueles que ajuda muito opensource ou ONG (desculpem :\). Mas esses dias me deu tal coceira pra querer aprender algo novo... e tal sentimento estava tão incômodo que até me fez deixar de lado os jogos por um tempo :(

Mesmo que o projeto fosse só pra estudar, tentei sempre fazer alguma funcionalidade por noite. E durante o dia (fora do trabalho nos trânsitos da vida) ficava fazendo anotações de próximos passos num caderninho. Claro, acho que só fui motivado pelo fato da curiosidade sobre um assunto e do incômodo anterior. Mas sempre tem aquela coisa: vai lendo Medium etc, acha um assunto interessante e aquilo fica te martelando e um estudo levando a outro.

Caso esses use cases genéricos ou projetos existentes não te deixar motivado, procure algum problema que você tem/teve/alguém teve no trabalho (ou fora) e que ninguém te deu oportunidade pra ir a fundo. Ou talvez ver como seria fazer algo que vc já fez, mas usando outro framework/tecnologia. Ou seja: definir um objetivo e um plano de ação.

**Comment by andersoonweb** [comment]
Boa, @kvnol! Seguinte, a ideia é chegar próximo ao real a experiência vai ser perfeita para aprender. Sendo assim, peguei uma ideia arquivada e coloquei em pratica. Eu sabia muita teoria, mas na pratica era outros quinhentos. Lembro-me que lia bastante o The sass way e quando fui aplicar, foi um terror! Erro pra tudo conte canto. Sem falar no Jekyll + Gulp. Mas foi muito massa, aprendi DEMAIS! E se for open, melhor ainda quando uma galera chega junto.
Sugiro que vc anote alguns pensamentos básicos e vai desenvolvendo. Hoje, se soubesse criar app p desktop eu faria um bloco de notas turbinado ao meu gosto, tipo do mac. Iria escolher Electon, ia ter que aprender mais de JS enfim, uma serie de coisas. E o melhor, num cenário real e útil.
Reinventar a roda às vezes é bastante benéfico.

Já escrevi sobre [https://medium.com/@andersoonweb/relato-de-uma-projeto-open-source-925ab40098e8](url)

Adendo, quando escrevi meu texto não tinha muita noção de escrita.
